extensions/panache/hibernate-reactive-panache/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/PanacheRepositoryBase.java

```diff
@@ -107,7 +108,9 @@ public default boolean isPersistent(Entity entity) {
      */
     @CheckReturnValue
     public default Uni<Void> flush() {
-        return INSTANCE.flush();
+        // TODO Luca not sure about this should it flush all the sessions?
+        // TODO Luca this will crash
```

Comment:
Good question :-/
Yeah, flush all the sessions.

If needed, add an overload for flushing a single PU?

Reply:
Actually… a repository targets a single entity, so it should flush that entity's PU by default, no?

Reply:
@FroMage yes but nowhere we save in the repository the type of the entity if I'm not mistaken and it gets erased at runtime. If you think we can find the type of `ENTITY` somewhere can you please me point me how?

Reply:
Again, with `@GenerateBridge` everything fits in (many many changes after :))